pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java

```diff
@@ -450,10 +450,14 @@ public static MethodType selectAmongMaximallySpecific(MethodType first, MethodTy
         Class<?> contextClass = context.getType();
 
         // search the class
-        for (Method method : contextClass.getDeclaredMethods()) {
-            if (isMethodApplicable(method, methodName, argArity, accessingClass, typeArguments)) {
-                result.add(getTypeDefOfMethod(context, method, typeArguments));
+        try {
+            for (Method method : contextClass.getDeclaredMethods()) {
```

Comment:
Isn't that already too late? I'd expect, that the LinkageError occurs the first time, we have the class instance (the classloader tried to load the class) - or do we get a partially linked class back, and we get the exception when retrieving the methods?

Reply:
Yes, it can happen at this point (actually, it occurred when analyzing Spring Framework, that's how I found out). Class graph is loaded lazily (that is, when class A uses B, which uses C, these are loaded as needed upon first access). When trying to load the classes required by the methods for args / return types, the runtime would detect they were missing and throw a `NoClassDefError`.